server/src/main/java/org/elasticsearch/cluster/metadata/IndexMetadata.java

```diff
@@ -1499,7 +1500,7 @@ private static class IndexMetadataDiff implements Diff<IndexMetadata> {
         private final IndexMetadataStats stats;
         private final Double indexWriteLoadForecast;
         private final Long shardSizeInBytesForecast;
-        private final Diff<Map<String, Set<String>>> fieldsForModels;
+        private final Diff<Map<String, Map<String, List<String>>>> fieldsForModels;
```

Comment:
Shouldn't we capture this as a specific record with:
- model ID
- destination field 
- List of optional source fields

Having a `<Map<String,  Map<String, List<String>>>` is a bit confusing to me.

No need to change it now, just a thought for accommodating changes to the structure.

Reply:
😵 Made it a little hard to read, but I think this makes sense 👍